document/document.go

```diff
@@ -34,8 +34,17 @@ func NewDocument() *Document {
 }
 
 // NewDocumentOf creates a new document and initializes it with the content of the provided object.
-// It returns nil if the object cannot be converted to a valid Document.
+// It returns nil if the object cannot be converted to a valid Document. it's necessary to specify
```

Comment:
I would rather extend the comment of the `Save()` function rather than this one.